server/remote_cache/chunked_manifest/chunked_manifest.go

```diff
@@ -0,0 +1,214 @@
+package chunked_manifest
+
+import (
+	"context"
+	"encoding/hex"
+	"io"
+	"strconv"
+
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/digest"
+	"github.com/buildbuddy-io/buildbuddy/server/util/proto"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	rspb "github.com/buildbuddy-io/buildbuddy/proto/resource"
+)
+
+const (
+	chunkedManifestPrefix = "_bb_chunked_manifest_v1_/"
+	chunkOutputFilePrefix = "chunk_"
+)
+
+type ChunkedManifest struct {
+	BlobDigest     *repb.Digest
+	ChunkDigests   []*repb.Digest
+	InstanceName   string
+	DigestFunction repb.DigestFunction_Value
+}
+
+func FromSplitBlobResponse(res *repb.SplitBlobResponse, blobDigest *repb.Digest, instanceName string, digestFunction repb.DigestFunction_Value) *ChunkedManifest {
+	return &ChunkedManifest{
+		BlobDigest:     blobDigest,
+		ChunkDigests:   res.GetChunkDigests(),
+		InstanceName:   instanceName,
+		DigestFunction: digestFunction,
+	}
+}
+
+func (cm *ChunkedManifest) ToSplitBlobResponse() *repb.SplitBlobResponse {
+	return &repb.SplitBlobResponse{
+		ChunkDigests: cm.ChunkDigests,
+	}
+}
+
+func (cm *ChunkedManifest) ToSpliceBlobRequest() *repb.SpliceBlobRequest {
+	return &repb.SpliceBlobRequest{
+		BlobDigest:     cm.BlobDigest,
+		ChunkDigests:   cm.ChunkDigests,
+		InstanceName:   cm.InstanceName,
+		DigestFunction: cm.DigestFunction,
+	}
+}
+
+// Store saves the chunked manifest to the Action Cache using the blob digest as
+// the AC key.
+func (cm *ChunkedManifest) Store(ctx context.Context, cache interfaces.Cache) error {
+
+	// Verify the chunks exist and match the expected blob digest.
+	// This is expensive, but prevents a malicious or corrupted manifest from being stored.
+	//
+	// TODO: This can be done in the background before Set is called, and this can be
```

Comment:
agree that this could run in a separate go routine, though the remaining of this function seems very light weight.

Reply:
Now with find missing also being called, running this in parallel could be more of a perf gain. Updated to use a errgroup